packages/core/src/test/shared/sam/sync.test.ts

```diff
@@ -909,140 +924,997 @@ describe('SyncWizard', async () => {
+
+            assert(parameters)
+
+            assert.strictEqual(parameters.template.uri.fsPath, templateFile.fsPath)
+            assert.strictEqual(parameters.projectRoot.fsPath, projectRoot.fsPath)
+            assert.strictEqual(parameters.paramsSource, ParamsSource.SamConfig)
+            assert.strictEqual(parameters.deployType, 'infra')
+            assert(!parameters.region)
+            assert(!parameters.stackName)
+            assert(!parameters.bucketSource)
+            assert(!parameters.syncFlags)
+            assert.strictEqual(parameters.skipDependencyLayer, true)
+        })
+    })
+})
+
+describe('SAM Sync', () => {
+    let sandbox: sinon.SinonSandbox
+    let testFolder: TestFolder
+    let projectRoot: vscode.Uri
+    let workspaceFolder: vscode.WorkspaceFolder
+    let templateFile: vscode.Uri
+
+    let mockGetSpawnEnv: sinon.SinonStub
+    let mockGetSamCliPath: sinon.SinonStub
+    let mockChildProcessClass: sinon.SinonStub
+    let mockSamSyncChildProcess: sinon.SinonStub
+
+    let spyWriteSamconfigGlobal: sinon.SinonSpy
+    let spyRunInterminal: sinon.SinonSpy
+
+    let mockDefaultCFNClient: sinon.SinonStubbedInstance<DefaultCloudFormationClient>
+    let mockDefaultS3Client: sinon.SinonStubbedInstance<DefaultS3Client>
+
+    // Dependency clients
+    beforeEach(async function () {
+        testFolder = await TestFolder.create()
+        projectRoot = vscode.Uri.file(testFolder.path)
+        workspaceFolder = getWorkspaceFolder(testFolder.path)
+        sandbox = sinon.createSandbox()
+
+        // Create template.yaml in temporary test folder and add to registery
+        templateFile = vscode.Uri.file(await testFolder.write('template.yaml', validTemplateData))
+        await (await globals.templateRegistry).addItem(templateFile)
+
+        // Simulate return of deployed stacks
+        mockDefaultCFNClient = sandbox.createStubInstance(CloudFormationClientModule.DefaultCloudFormationClient)
+        sandbox.stub(CloudFormationClientModule, 'DefaultCloudFormationClient').returns(mockDefaultCFNClient)
+        mockDefaultCFNClient.listAllStacks.returns(intoCollection(stackSummaries))
+
+        // Simulate return of list bucket
+        mockDefaultS3Client = sandbox.createStubInstance(S3ClientModule.DefaultS3Client)
+        sandbox.stub(S3ClientModule, 'DefaultS3Client').returns(mockDefaultS3Client)
+        mockDefaultS3Client.listBucketsIterable.returns(intoCollection(s3BucketListSummary))
+
+        // Create Spy for validation
+        spyWriteSamconfigGlobal = sandbox.spy(SamConfigModule, 'writeSamconfigGlobal')
+        spyRunInterminal = sandbox.spy(ProcessTerminalUtils, 'runInTerminal')
+
+        // generate template.yaml in temporary test folder and add to registery
+        templateFile = vscode.Uri.file(await testFolder.write('template.yaml', validTemplateData))
+        await (await globals.templateRegistry).addItem(templateFile)
```

Comment:
We are adding things to globals but they are not removed after test. This could potentially affect other tests.

Reply:
No side effect seen, but there is potential. I'll add clean up step.

Reply:
Added step to reset registry.